Builder/ParametersGenerator.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Becklyn\RouteTreeBundle\Builder;
+
+use Becklyn\RouteTreeBundle\Tree\Node;
+
+
+/**
+ *
+ */
+class ParametersGenerator
+{
+    /**
+     * Automatically sets the parameters for all descendant nodes
+     *
+     * @param Node $node
+     */
+    public function generateParametersForNode (Node $node)
+    {
+        $node->setParameters($this->calculateParametersForNode($node));
```

Comment:
Wäre eine Situation realistisch in dem ein Node schon Parameter gesetzt hat, die hier verloren gehen würden?